MOOSE triggers compiler warnings with newer clang++ and newer C++ standards

### Bug Description

I want to use a recent Kokkos version requiring C++20.  I want to turn on `-Werror` so I can catch any compiler warnings along the way.  MOOSE fails to build with `clang++` 20.1.8 `-std=c++20` (or newer) with `-Werror` turned on, with e.g.
```
/tmp/build/moose/framework/contrib/hit/src/hit/lex.cc:16:19: error: unused variable 'newline' [-Werror,-Wunused-const-variable]
   16 | const std::string newline = "\n\r";
```

### Steps to Reproduce

See e.g. https://civet.inl.gov/job/3859397/; any build with a new enough `clang++ -std=c++20 -Werror` (or a higher standard) seems to be sufficient.

### Impact

Just an annoyance for now, but it may become a bigger deal when we start requiring C++20 for everyone.

### [Optional] Diagnostics

_No response_

--- a/framework/contrib/gtest/gtest/gtest-printers.h
+++ b/framework/contrib/gtest/gtest/gtest-printers.h
@@ -521,11 +521,11 @@ GTEST_API_ void PrintTo(wchar_t wc, ::std::ostream* os);
 
 GTEST_API_ void PrintTo(char32_t c, ::std::ostream* os);
 inline void PrintTo(char16_t c, ::std::ostream* os) {
-  PrintTo(ImplicitCast_<char32_t>(c), os);
+  PrintTo(static_cast<char32_t>(c), os);
 }
 #ifdef __cpp_lib_char8_t
 inline void PrintTo(char8_t c, ::std::ostream* os) {
-  PrintTo(ImplicitCast_<char32_t>(c), os);
+  PrintTo(static_cast<char32_t>(c), os);
 }
 #endif
 

--- a/framework/contrib/hit/src/hit/lex.cc
+++ b/framework/contrib/hit/src/hit/lex.cc
@@ -13,7 +13,6 @@ const std::string digits = "0123456789";
 const std::string alpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 const std::string space = " \t";
 const std::string allspace = " \t\n\r";
-const std::string newline = "\n\r";
 const std::string alphanumeric = digits + alpha;
 const std::string identchars = alphanumeric + "_./:<>-+*!";
 

